### ISSUE TITLE
Incorrect mapping over `this.props.widgetsFor` in Custom preview

### ISSUE THREAD

**Issue by crgeary** [decaporg/decap-cms#1089]
**- Do you want to request a *feature* or report a *bug*?**

BUG

**- What is the current behavior?**

The following code within a PreviewTemplate outputs the words `data` & `widgets` when there are no images to map over.
```
this.props.widgetsFor('images').map(function (image, index) {

});
```

**- If the current behavior is a bug, please provide the steps to reproduce.**

1. Visit - https://practical-leavitt-aada1c.netlify.com/admin/
2. Click "Login" (there is no password)
3. Click "New Pages"
4. View the Preview

**- What is the expected behavior?**

I would expect the above block of code to run per "image" in the "images" data. If there are no images, then there should be nothing to map over.

**- Please mention your CMS, node.js, and operating system version.**

Netlify CMS v1.1

**- Please link or paste your `config.yml` below if applicable.**

I've made [a repo](https://github.com/crgeary/netlify-cms-widgetsfor-issue) containing as a little as possible to show the issue:
 - [/static/admin/config.yml](https://github.com/crgeary/netlify-cms-widgetsfor-issue/blob/master/static/admin/config.yml)
 - [/static/admin/cms.js](https://github.com/crgeary/netlify-cms-widgetsfor-issue/blob/master/static/admin/cms.js)

* * *

Please let me know if there is anything more I can do to help.

**Comment by erquhart** [comment]
Seems to only happen on initial load - if you add an item to the list and then remove it, the preview text is blank.

**Comment by mdavidmorton** [comment]
Ran into this same problem today. Solution below probably isn't great long term, but it bypasses the `data widgets` text:

```
/* Assign to variable before render() return statement */
var images = this.props.widgetsFor('images'); 

/* Other code */

h('div', {'className': 'band band--md'}, (images.first() != undefined)
    ? images.map(function (image, index) {        
        return h('div', {'key': index},
            this.props.getAsset(image.getIn(['data', 'src'])).toString()
        );
      })
    : ''
)
```

**Comment by tech4him1** [comment]
Adding a default value to the list widget should fix this (#977).

**Comment by martinjagodic** [comment]
Closing as stale and/or outdated.
